src/Curl/Curl.php

```diff
@@ -1100,4 +1123,196 @@ public static function is_array_multidim($array)
+     */
+    public function getBeforeSendFunction()
+    {
+        return $this->beforeSendFunction;
+    }
+
+    /**
+     * @return null
+     */
+    public function getCompleteFunction()
+    {
+        return $this->completeFunction;
+    }
+
+    /**
+     * @return resource
+     */
+    public function getCurl()
+    {
+        return $this->curl;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isCurlError()
+    {
+        return $this->curlError;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCurlErrorCode()
+    {
+        return $this->curlErrorCode;
+    }
+
+    /**
+     * @return null
+     */
+    public function getCurlErrorMessage()
+    {
+        return $this->curlErrorMessage;
+    }
+
+    /**
+     * @return null
+     */
+    public function getDownloadCompleteFunction()
+    {
+        return $this->downloadCompleteFunction;
+    }
+
+    /**
+     * @return null
+     */
+    public function getEffectiveUrl()
+    {
+        return $this->effectiveUrl;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isError()
+    {
+        return $this->error;
+    }
+
+    /**
+     * @return int
+     */
+    public function getErrorCode()
+    {
+        return $this->errorCode;
+    }
+
+    /**
+     * @return null
+     */
+    public function getErrorFunction()
+    {
+        return $this->errorFunction;
+    }
+
+    /**
+     * @return null
+     */
+    public function getErrorMessage()
+    {
+        return $this->errorMessage;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isHttpError()
+    {
+        return $this->httpError;
+    }
+
+    /**
+     * @return null
+     */
+    public function getHttpErrorMessage()
+    {
+        return $this->httpErrorMessage;
+    }
+
+    /**
+     * @return int
+     */
+    public function getHttpStatusCode()
+    {
+        return $this->httpStatusCode;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return null
+     */
+    public function getRawResponse()
+    {
+        return $this->rawResponse;
+    }
+
+    /**
+     * @return string
+     */
+    public function getRawResponseHeaders()
+    {
+        return $this->rawResponseHeaders;
+    }
+
+    /**
+     * @return null
+     */
+    public function getRequestHeaders()
+    {
+        return $this->requestHeaders;
+    }
+
+    /**
+     * @return null
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
+
+    /**
+     * @return null
+     */
+    public function getResponseHeaders()
+    {
+        return $this->responseHeaders;
+    }
+
+    /**
+     * @return null
+     */
+    public function getSuccessFunction()
+    {
+        return $this->successFunction;
+    }
+
+    /**
+     * @return null
+     */
+    public function getUrl()
+    {
+        return $this->url;
+    }
```

Comment:
Do you need all these methods? For example `completeFunction` is used internally, but made public to be reused internally.